app/views/filters/_form.html.erb

```diff
@@ -42,9 +42,16 @@
           </span>
       </div>
 
-      <div id="granular_form" style="<%= !f.object.granular? ? 'display:none' : '' %>">
+      <div id="override_taxonomy_form" class="<%= f.object.allows_taxonomies_filtering? ? '' : 'hidden' %>">
+        <%= checkbox_f f, :override?,
+                       :help_inline => _("Filters inherit organizations and locations from their role by default. If override field is enabled, filter can override the set of its organizations and locations. Later role changes will not affect such filter. After disabling override field, role organizations and locations apply again."),
+                       :id => 'override_taxonomy_checkbox' %>
+      </div>
+
+      <div id="granular_form" style="<%= f.object.granular? ? '' : 'display:none' %>">
         <%= checkbox_f f, :unlimited?,
-                       :help_inline => _("By unchecking this you can specify filter using Foreman search syntax in search field. If filter remains unlimited (this field is checked) it applies on all resources of the selected type") %>
+                       :help_inline => _("By unchecking this you can specify filter using Foreman search syntax in search field. If filter remains unlimited (this field is checked) it applies on all resources of the selected type. When role is associated with some organizations or locations filter is not considered unlimited as it's scoped accordingly.") %>
```

Comment:
Not sure what you mean by ` if locations are disabled, role is not associated with it right` - I just mean the same thing as in the other comment, the text is dependent on both organizations/locations being available, but we can't assume that. Notice `:unlimited?` is shown even when there are no locs/orgs